cpp/src/lists/contains.cu

```diff
@@ -44,296 +41,231 @@
-
     CUDF_EXPECTS(!cudf::is_nested(lists.child().type()),
                  "Nested types not supported in list search operations.");
-    CUDF_EXPECTS(lists.child().type() == search_key.type(),
+    CUDF_EXPECTS(lists.child().type() == search_keys.type(),
                  "Type/Scale of search key does not match list column element type.");
-    CUDF_EXPECTS(search_key.type().id() != type_id::EMPTY, "Type cannot be empty.");
+    CUDF_EXPECTS(search_keys.type().id() != type_id::EMPTY, "Type cannot be empty.");
 
     auto constexpr search_key_is_scalar = std::is_same_v<SearchKeyType, cudf::scalar>;
-
-    if constexpr (search_keys_have_nulls && search_key_is_scalar) {
-      return make_numeric_column(data_type(type_id::INT32),
+    if (search_key_is_scalar && search_keys_have_nulls) {
+      // If the scalar key is invalid/null, the entire output column will be all nulls.
+      return make_numeric_column(data_type{cudf::type_to_id<size_type>()},
                                  lists.size(),
                                  cudf::create_null_mask(lists.size(), mask_state::ALL_NULL, mr),
                                  lists.size(),
                                  stream,
                                  mr);
     }
 
-    auto const device_view = column_device_view::create(lists.parent(), stream);
-    auto const d_lists     = lists_column_device_view{*device_view};
-    auto const d_skeys     = get_search_keys_device_iterable_view(search_key, stream);
-
-    auto result_positions = make_numeric_column(
-      data_type{type_id::INT32}, lists.size(), cudf::mask_state::UNALLOCATED, stream, mr);
-    auto result_validity = make_numeric_column(
-      data_type{type_id::BOOL8}, lists.size(), cudf::mask_state::UNALLOCATED, stream, mr);
-    auto mutable_result_positions =
-      mutable_column_device_view::create(result_positions->mutable_view(), stream);
-    auto mutable_result_validity =
-      mutable_column_device_view::create(result_validity->mutable_view(), stream);
-    auto search_key_iter =
-      cudf::detail::make_pair_rep_iterator<ElementType, search_keys_have_nulls>(*d_skeys);
-
-    search_each_list_row<ElementType>(d_lists,
-                                      search_key_iter,
-                                      find_option,
-                                      *mutable_result_positions,
-                                      *mutable_result_validity,
-                                      stream);
-
-    auto [null_mask, num_nulls] = construct_null_mask(lists, result_validity->view(), stream, mr);
-    result_positions->set_null_mask(std::move(null_mask), num_nulls);
-    return result_positions;
+    auto const lists_cdv_ptr = column_device_view::create(lists.parent(), stream);
+    auto const lists_cdv     = cudf::detail::lists_column_device_view{*lists_cdv_ptr};
+
+    auto out_positions = make_numeric_column(
+      data_type{type_to_id<size_type>()}, lists.size(), cudf::mask_state::UNALLOCATED, stream, mr);
```

Comment:
(For a future issue/PR) We should add a `column` constructor to be constructible from a `device_uvector&&` so for outputs like this we can just more conveniently construct a `device_uvector<size_type>` and then construct the column when returning the output:
```
rmm::device_uvector<size_type> out_positions{...};
...
return make_unique<column>{std::move(out_positions)};
```
```

Reply:
Issue filed: https://github.com/rapidsai/cudf/issues/11115